source/jormungandr/jormungandr/instance_manager.py

```diff
@@ -256,15 +235,19 @@ def _find_coverage_by_object_id_in_instances(self, instances, object_id):
     def _all_keys_of_id_in_instances(self, instances, object_id):
         valid_instances = []
         for instance in instances:
-            if self._exists_id_in_instance(instance, object_id):
+            if self._exists_id_in_instance(instance.name, instance.publication_date, object_id):
                 valid_instances.append(instance)
         if not valid_instances:
             raise RegionNotFound(object_id=object_id)
 
         return valid_instances
 
     @cache.memoize(app.config[str('CACHE_CONFIGURATION')].get(str('TIMEOUT_PTOBJECTS'), None))
-    def _exists_id_in_instance(self, instance, object_id):
+    def _exists_id_in_instance(self, instance_name, instance_publication_date, object_id):
+        """
+        published_date is used to invalidate the cache when updating the ntfs, poi
```

Comment:
You've got an extra `, poi` in you comment anyway, I think.

Also I thought of a "clearer" naming idea:
```suggestion
    def _exists_id_in_instance(self, instance_name, extra_cache_key, object_id):
        """
        instance's published_date is usually provided as extra_cache_key to invalidate the cache when updating the ntfs
```